File: backend/schemas/analytics.py
"""Pydantic schemas for analytics API responses."""

from pydantic import BaseModel, Field
from typing import List, Dict


class UsageStats(BaseModel):
    """Usage statistics summary."""

    query_count: int = Field(ge=0, description="Total number of queries")
    document_count: int = Field(ge=0, description="Total number of documents")
    session_count: int = Field(ge=0, description="Total number of chat sessions")


class ActivityPoint(BaseModel):
    """Activity data point representing queries on a specific date."""

    date: str = Field(description="Date in YYYY-MM-DD format")
    count: int = Field(ge=0, description="Number of queries")


class DocumentTypeStats(BaseModel):
    """Breakdown of documents by file type."""

    pdf: int = Field(0, ge=0)
    docx: int = Field(0, ge=0)
    txt: int = Field(0, ge=0)


class AnalyticsResponse(BaseModel):
    """Full dashboard analytics response."""

    stats: UsageStats
    activity_timeline: List[ActivityPoint]
    document_types: DocumentTypeStats
    storage_used: int = Field(ge=0, description="Total storage used in bytes")

File: frontend/lib/api/analytics.ts
import { apiClient } from './client';
import type { AnalyticsDashboard } from '@/types/document';

export const analyticsApi = {
  getDashboard: async (period: '7d' | '30d' = '7d'): Promise<AnalyticsDashboard> => {
    const response = await apiClient.get('/api/analytics/dashboard', {
      params: { period },
    });
    const data = response.data;

    return {
      stats: {
        queryCount: data.stats.query_count !== undefined ? data.stats.query_count : data.stats.queryCount,
        documentCount: data.stats.document_count !== undefined ? data.stats.document_count : data.stats.documentCount,
        sessionCount: data.stats.session_count !== undefined ? data.stats.session_count : data.stats.sessionCount,
      },
      activityTimeline: data.activity_timeline || data.activityTimeline || [],
      documentTypes: data.document_types || data.documentTypes || { pdf: 0, docx: 0, txt: 0 },
      storageUsed: data.storage_used !== undefined ? data.storage_used : data.storageUsed,
    };
  },
};
